c/extras/dt_ftrl.cc

```diff
@@ -64,33 +64,28 @@ void Ftrl::fit(const DataTable* dt_X, const DataTable* dt_y) {
   auto c_y = static_cast<BoolColumn*>(dt_y->columns[0]);
   auto d_y = c_y->elements_r();
 
-  // Do training for `nepochs`.
-  for (size_t i = 0; i < params.nepochs; ++i) {
-    double total_loss = 0;
+  const RowIndex ri_X = dt_X->rowindex;
```

Comment:
@st-pasha, from what I understand, rowindex is deprecated for `DataTable`, right? Then which column rowindex one should use here and also for `predict`?

Reply:
This is correct. DataTable's rowindex will get removed in the near future, and should not be used. Each column may have its own rowindex. You could detect and process specially the case when all columns have the same rowindex (it is quite common), but, in general, different columns are allowed to have different rowindices.

Reply:
I see. So when we do something like `df[r1:r2,:]`, all columns will have the same rowindex, but if, for instance, we sort by two columns it may not be the case?

Reply:
When sorting by 2 columns the rowindex will also be the same.
You can get different rowindices by: cbinding 2 frames that have different rowindices; selecting from a Frame some regular columns + some computed columns, e.g. `DT[:100, [f.A, f.A+f.B]]`; performing a join; etc.

Reply:
I see, if rowindices can be different for different columns, iterating over just one will not work...